### PR TITLE
Implement `verdi config`

### LINKED ISSUE #2208: replace `verdi devel setproperty` by `verdi config`
- [x] convert 'verdi devel setproperty' to 'verdi config'
 - [x] Rename 'properties' to 'options'
 - [x] mirror the API from "git config"

### PR DESCRIPTION
**Pull request by sphuber**
Fixes #2208 

The new command `verdi config` replaces the various commands under
`verdi devel` that were used to list, get, set and unset configuration
options, that were used to be called "properties". Since the term
"property" is a reserved keyword, it was decided to rename them to
"options", which is also the term used by "git". The interface of
`verdi config` also mirrors that of `git config`. That is to say
to get the value of an option simply call `git config <option_name>`
and to set it `git config <option_name> <option_value>`. To unset
it, the `--unset` flag can be used. Finally, the getting, setting
or unsetting can be applied to a certain "scope", meaning to be
configuration wide or profile specific.

To make the implementation of this command simple and clean, the
bulk of the work was in refactoring the definition, construction and
operation on the configuration of an AiiDA instance. This is now
represented by the `Config` class, through which these configuration
options can be set or unset as well as retrieved.

### CODE REVIEW

**Comment by sphuber** [comment]
The mock profile function currently just generates a dummy profile, with dummy variables. The profile is not "actually" used. The only shared part might be the repository path, but since the `aiida.backends.tests.utils.configuration.with_temporary_config_instance` decorator creates a completely temporary directory in which everything is created, these should never overlap.

Note that currently the creation of a new "profile" is still done manually by the create config functions that are called by quicksetup and setup, i.e. they manually build a dictionary with the required keys. In a future PR this should be refactored, as the Profile class is the only one that should now about these keys.

File: docs/source/verdi/verdi_user_guide.rst [hunk 1]
```diff
@@ -355,12 +355,7 @@ Manage ``Data`` nodes.
 ---------------
 Commands intended for developers, such as setting :doc:`config properties<properties>` and running the unit test suite.
 
-  * **delproperty**: delete a property from the configuration
```
Comment (giovannipizzi) [inline docs/source/verdi/verdi_user_guide.rst]: Are the new command line verdi commands described? (couldn't see the change)

Comment (sphuber) [reply docs/source/verdi/verdi_user_guide.rst]: I have added the new command to the `verdi` docs

**Comment by sphuber** [comment]
And to answer your other question, no this does not change the functioning of `AIIDA_PATH`, it has just been moved to `aiida.manage.configuration.settings`.

**Pull request merged**